ObstacleAvoidance.ino

```diff
@@ -1,49 +1,43 @@
 #include <Smartcar.h>
 
+//Initialize different Pin value
 int leftMotorForwardPin = 8;
-
 int leftMotorBackwardPin = 10;
-
 int leftMotorSpeedPin = 9;
-
 int rightMotorForwardPin = 12;
-
 int rightMotorBackwardPin = 13;
-
 int rightMotorSpeedPin = 11;
+const int TRIGGER_PIN = 6; //D6
+const int ECHO_PIN = 7; //D7
+
+//Initialize magic value
+const unsigned int MAX_DISTANCE = 100;
+const int STOP_DISTANCE = 20;
 
 BrushedMotor leftMotor(leftMotorForwardPin,leftMotorBackwardPin, leftMotorSpeedPin);
 
 BrushedMotor rightMotor(rightMotorForwardPin, rightMotorBackwardPin, rightMotorSpeedPin);
 
 DifferentialControl control(leftMotor, rightMotor);
 
-SimpleCar car(control);
-
-const int TRIGGER_PIN = 6; //D6
-const int ECHO_PIN = 7; //D7
-const unsigned int MAX_DISTANCE = 100000;
 SR04 front(TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
 
+SimpleCar car(control);
 
 void setup() {
-Serial.begin(9600);
+ 
+ Serial.begin(9600);
+ 
  car.setSpeed(100);
  
-
-
-
 }
 
 void loop() {
-  
-Serial.println(front.getDistance());
-
-if(front.getDistance() < 20){
-car.setSpeed(0);
 
-}else{
-  car.setSpeed(100);
+ if(front.getDistance() < STOP_DISTANCE && front.getDistance() > 0){
```

Comment:
This will conduct two separate measurements that can potentially yield different results.
This means that if the first call of `getDistance` returns `0` and the second one `100` your car will stop despite there not being an obstacle nearby.